src/Avalonia.Base/Data/Core/PropertyAccessorNode.cs

```diff
@@ -60,16 +73,7 @@ protected override void StartListeningCore(WeakReference<object> reference)
                 }
             }
 
-            if (accessor == null)
-            {
-                reference.TryGetTarget(out object instance);
-
-                var message = $"Could not find a matching property accessor for '{PropertyName}' on '{instance}'";
-
-                var exception = new MissingMemberException(message);
-
-                accessor = new PropertyError(new BindingNotification(exception, BindingErrorType.Error));
-            }
+            Debug.Assert(accessor != null);
```

Comment:
I'd prefer to not have any `Debug.Assert`s in the codebase as it can cause message boxes to pop up during unit testing and cause havoc. I usually throw an `AvaloniaInternalException` instead.